ha/fault_tolerance/fault_monitor.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+"""
+Handler for handling events received from k8s monitor service
+"""
+
+
+import traceback
+from cortx.utils.conf_store import Conf
+from cortx.utils.log import Log
+from ha.util.message_bus import MessageBus, CONSUMER_STATUS, MessageBusConsumer
+from ha import const
+from ha.core.event_analyzer.event_analyzerd import EventAnalyzer
+from ha.k8s_setup.const import _DELIM
+
+from consul.base import ConsulException
+from cortx.utils.conf_store.error import ConfError
+from ha.core.event_analyzer.event_analyzer_exceptions import EventFilterException
+from ha.core.event_analyzer.event_analyzer_exceptions import EventParserException
+from ha.core.event_analyzer.event_analyzer_exceptions import SubscriberException
+
+class FaultMonitor:
+    """
+    Base class for all faults received from k8s monitor
+    Faults can be for host, node, container, pvc etc
+    Seperate derived class to be written for each of the above modules as and when required
+    """
+
+    def __init__(self):
+        """Init method"""
+        self._consumer = self._get_consumer()
+
+    def start(self):
+        """
+        Start to listen messages.
+        """
+        self._consumer.start()
```

Comment:
I think a check is needed here whether self._consumer is None or not.

Reply:
Done